packages/core/admin/admin/src/pages/MarketplacePage/tests/providers.test.js

```diff
@@ -333,4 +317,33 @@ describe('Marketplace page - providers tab', () => {
     );
     expect(downloadsLabel).toBeVisible();
   });
+
+  it('paginates the results', async () => {
+    // Should have pagination section with 4 pages
+    const pagination = screen.getByLabelText(/pagination/i);
+    expect(pagination).toBeVisible();
+    const pageButtons = screen.getAllByRole('link', { name: /go to page \d+/i });
+    expect(pageButtons.length).toBe(4);
+
+    // Can't go to previous page since there isn't one
+    expect(screen.getByRole('link', { name: /go to previous page/i })).toHaveAttribute(
```

Comment:
Is there anyway we could avoid the getByRoles? As we've seen they are costly

Reply:
getByRole was convenient here because it picks up the aria label. I'll try to see if it works with getByLabelText maybe. Otherwise I would add test IDs but I'm not a fan of that, especially since it's in the helper plugin

Reply:
I ended up using getByLabelTest with closest() to find the parent link. It saves about 5 seconds in total which is cool.

But I'm pretty bummed that we can't use getByRole, I like that API, it's a good way to make sure you write accessible code